src/python/WMCore/WMSpec/StdSpecs/Resubmission.py

```diff
@@ -74,25 +83,44 @@ def getWorkloadCreateArgs():
                     "SiteWhitelist": {"default": [], "type": makeList,
                                       "validate": lambda x: all([cmsname(y) for y in x])},
                     # it can be Chained or MC requests, so lets make it optional
-                    "InputDataset": {"optional": True, "validate": dataset, "null": True}}
+                    "InputDataset": {"optional": True, "validate": dataset, "null": True},
+                    ### Override StdBase parameter definition
+                    "TimePerEvent": {"default": None, "type": float, "null": True, "validate": checkTimePerEvent},
+                    "Memory": {"default": None, "type": float, "null": True, "validate": checkMemCore},
+                    "Multicore": {"default": None, "type": int, "null": True, "validate": checkMemCore},
+                    "EventStreams": {"default": None, "type": int, "null": True, "validate": checkEventStreams}
+                    }
 
         StdBase.setDefaultArgumentsProperty(specArgs)
         return specArgs
 
+    def fixupArguments(self, arguments, userArgs):
+        """
+        This method will ensure that:
+         * if the user provided some specific arguments, it will be passed down the chain
+         * otherwise, the same argument from the original/parent workflow will be dumped
+        The only arguments to be tweaked like that are:
+            TimePerEvent, Memory, Multicore, EventStreams
+        :param arguments: full set of arguments from creation+assignment definitions
+        :param userArgs: solely the key/value pair values provided by the client
+        :return: nothing, updates are made in place
+        """
+        specialArgs = ("TimePerEvent", "Memory", "Multicore", "EventStreams")
+        argsDefinition = self.getWorkloadCreateArgs()
+        for arg in specialArgs:
+            if arg not in userArgs:
+                arguments[arg] = argsDefinition[arg]["default"]
```

Comment:
From the recent validation in testbed:
* We should not set the default value (null) for `ReReco` requests.
* Memory and TimePerEvent for StepChain and ReReco should be a single value
* We should also update the json specification accordingly (meaning, if Memory for Task1 was overwritten to 1GB, the Task1 dictionary should have the correct value). HOWEVER, this one is definitely a different issue.